proto/ignition/msgs/oreinted_3d_box.proto

```diff
@@ -0,0 +1,45 @@
+
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package ignition.msgs;
+option java_package = "com.ignition.msgs";
+option java_outer_classname = "Oriented3DBoxProtos";
+
+/// \ingroup ignition.msgs
+/// \interface Oriented3DBox
+/// \brief A 3D oreinted bounding box
+
+import "ignition/msgs/header.proto";
+import "ignition/msgs/vector3d.proto";
+import "ignition/msgs/quaternion.proto";
+
+message Oriented3DBox
+{
+  /// \brief Optional header data
+  Header header = 1;
+
+  /// \brief Center of the bounding box
+  Vector3d center = 2;
+
+  /// \brief Orientation of the bounding box
+  Quaternion orientation = 3;
+
+  /// \brief The size of the bounding box (width/height/depth)
```

Comment:
We usually use this convention on 3D:

* X: forward
* Z: up

I believe this would make the size be:

```suggestion
  /// \brief The size of the bounding box on XYZ (depth/width/height)
```

Reply:
Actually i provide it in camera coord not world coord
@chapulina

Reply:
> Actually i provide it in camera coord not world coord

If that's the case, and @chapulina is okay with this being in camera coordinates, then would you mind adding a comment here that mentions this is in camera coordinates? However, I believe that the message types should be as generic as possible, so I wonder if it makes more sense to have it be in world coordinates instead of camera coordinates, and then updating code that uses this message type in other repositories to use world coordinates instead.

Reply:
ok, I want to clarify that the world coord that can be used is not the OGRE world coord ... but it must be a coordinates from the camera view, we should have the 3d boxes wrt the camera position and orientation

what we are talking about now, is just changing the axes by multiplying with the matrix that swap the axes
```
math::Matrix3d tf(0, 0, -1,
                       -1, 0,  0,
                        0, 1,  0);
```

Anyway, when we use that code to generate a 3D dataset, usually datasets have a calibration file that maps lidar coord to camera coord or vise versa, so I think we will put a matrix that swaps the axes for conversion

Reply:
Gazebo and ROS generally use X-forward, Z-up convention, even for cameras. So strictly speaking, if you're using the camera frame, the X is sticking into the image and represents depth.

Then besides that, there's an extra optical frame which is Z-forward, Y-down (like in most computer vision applications). If we're talking about the optical frame, then the original documentation is correct. 

[REP-103](https://www.ros.org/reps/rep-0103.html#id22) documents some frame conventions.

Camera frames can get confusing, so I think it's worth making the frames clear on the message documentation. I don't have a preference about the frame, as long as it's documented.

Reply:
Ok, I used the 3d camera coordinates and documented the msg with that.
Btw, I am trying to make the dataset generation to be in the same format of the famous datasets, and the 3d dataset am following is [KITTI dataset](http://www.cvlibs.net/datasets/kitti/eval_object.php?obj_benchmark=3d) and it provides the 3d boxes in 3d camera coordinates [KITTI format](https://github.com/bostondiditeam/kitti/blob/master/resources/devkit_object/readme.txt#L60)